keyboards/z150_blackheart/keymaps/default_tkl/keymap.c

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2018 blindassassin111
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_z150_tkl(
+        KC_F1,   KC_F2,     KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,     KC_INS,  KC_HOME, KC_PGUP,
+        KC_F3,   KC_F4,     KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,     KC_DEL,  KC_END,  KC_PGDN,
+        KC_F5,   KC_F6,     KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,               XXXXXXX, XXXXXXX, XXXXXXX,
+        KC_F7,   KC_F8,     KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),       XXXXXXX, KC_UP,   XXXXXXX,
+        KC_F9,   KC_F10,    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_MENU, KC_RCTL,     KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT_z150_tkl(
+        _______, _______,   RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, _______, _______,
+        _______, _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, _______, _______,
+        _______, _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,              _______, _______, _______,
+        _______, _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, _______, _______,
+        _______, _______,   _______, _______, _______,                            _______,                            _______, _______, _______, _______,     _______, _______, _______
+    ),
+};
+
+void matrix_init_user(void) {
+}
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+    setPinOutput(B0);
```

Comment:
The `setPinOutput`'s should be in the `matrix_init_user` function, not here.

Reply:
Any insight to elevating these to keyboard level (`matrix_init_kb`, supposedly)?

Reply:
I'm personally not a fan of doing it, because it makes it harder for the user to override. 
The flip side, is that moving it there means that it is functional when using the QMK Configurator.

So.... I think it would be best to move to the keyboard level, yeah.

Reply:
All these overrideable functions should really be doing something similar to `process_record_*()`, returning a boolean which the higher-level functions use to determine whether or not to proceed with their own code. But changing all those signatures would touch just about every keyboard :(

Reply:
Probably. Though, that's a massive change...  But it may be worth doing, long term.

Reply:
But for now, yeah, I think the led code should be moved to the kb functions, instead.